pages/_app.tsx

```diff
@@ -34,4 +41,15 @@ function ElliotApp({ Component, pageProps }: AppProps) {
   )
 }
 
+// Getting cookies from the request on every server-side render.
+// Warning: This disables the ability to perform automatic static optimization.
```

Comment:
it disable it in all the pages, correct?
does it means it doesn't make so much sense to use getInitialProps or getStaticProps in pages anymore?
instead is better to use getServerSideProps everywhere?

Reply:
Yes, on all pages, because `_app` is called for all pages. And from here we call the pages' `getInitialProps` so I'd recommend to only use `getInitialProps` in pages. Mixing the approaches would be confusing.